packages/beacon-node/src/chain/chain.ts

```diff
@@ -567,6 +571,68 @@ export class BeaconChain implements IBeaconChain {
     if (headState) {
       this.opPool.pruneAll(headState);
     }
+
+    if (!this.opts.disableArchiveOnCheckpoint) {
+      // Archive state and blocks
+      // TODO: When is the best time to perform this actions?
+      // TODO: Only perform one archive at a time
```

Comment:
Doesn't this reintroduce https://github.com/ChainSafe/lodestar/issues/2015?

Reply:
It should not because all fork-choice ops are done before any async call in `archiveFinalizedStateAndBlocks`. So because we have a queue that ensures only one block is imported at once, fork-choice .prune() won't be called with stale data